smart_open/ssh.py

```diff
@@ -43,18 +44,22 @@
 DEFAULT_PORT = 22
 
 
-def _connect(hostname, username, port):
+def _connect(hostname, username, port, rsakey = None):
     key = (hostname, username)
     ssh = _SSH.get(key)
     if ssh is None:
         ssh = _SSH[key] = paramiko.client.SSHClient()
         ssh.load_system_host_keys()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(hostname, port, username)
+        if rsakey:
+            sshkey = paramiko.RSAKey.from_private_key(StringIO(rsakey))
+            ssh.connect(hostname, port, username, pkey = sshkey)
+        else:
+            ssh.connect(hostname, port, username)
     return ssh
 
 
-def open(path, mode='r', host=None, user=None, port=DEFAULT_PORT):
+def open(path, mode='r', host=None, user=None, port=DEFAULT_PORT, transport_params={}):
```

Comment:
Please have a look at how other submodules handle transport_params.

Reply:
In the light of https://github.com/RaRe-Technologies/smart_open/issues/304 `transport_params={}` most certainly should be replaced with 
```py
def open(..., transport_params=None):
if transport_params is None:
     transport_params = {}
# end if
```